bigchaindb/backend/schema.py

```diff
@@ -99,3 +105,21 @@ def init_database(connection=None, dbname=None):
     create_database(connection, dbname)
     create_tables(connection, dbname)
     create_indexes(connection, dbname)
+
+
+def validate_if_exists_asset_language(tx_body):
+    data = tx_body['asset'].get('data', {})
+
+    if data and ('language' in data):
+
+        language = data.get('language')
```

Comment:
This only checks the `language` is set in the top level document. With MongoDB the `language` can be overridden in embedded documents.

Something like:
```python
asset={'msg': {'language': 'hello'}}
```
will pass the check but will still fail when writing to mongodb.

Reply:
@kansi This comment remains unresolved.

Reply:
@ttmc   @r-marques is right. we will need to interate over all the nested keys.